ovos_plugin_manager/utils/ui.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+from ovos_utils.log import LOG
+
+from ovos_plugin_manager.stt import get_stt_lang_configs
+from ovos_plugin_manager.tts import get_tts_lang_configs
+
+
+def hash_dict(d):
+    return str(hash(json.dumps(d, indent=2, sort_keys=True, ensure_ascii=True)))
+
+
+class PluginGUIHelper:
+    """ Helper class to provide metadata for GUI consumption
+    This allows all sorts of rich integrations by
+    any downstream application wanting to provide a plugin store
+
+    This is the central place to manage anything GUI related,
+    downstream should not need to import anything else
+    """
+    _stt_opts = {}
+    _tts_opts = {}
+
+    @classmethod
+    def stt_option2config(cls, opt):
+        """ get the equivalent plugin config from a GUI display model """
+        return cls._stt_opts.get(hash_dict(opt))
+
+    @classmethod
+    def stt_config2option(cls, cfg, lang=None):
+        """ get the equivalent GUI display model from a plugin config """
+        engine = cfg["engine"]
```

Comment:
TODO - i dont think "engine" is always present, verify the other helper methods to ensure it is